admin/services/school-impersonation.service.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+const R = require('ramda')
+
+const schoolDataService = require('../services/data-access/school.data.service')
+const schoolImpersonationEmptyValueValidator = require('../lib/validator/school-impersonation/school-impersonation-empty-value-validator')
+const schoolImpersonationDfeNumberValidator = require('../lib/validator/school-impersonation/school-impersonation-dfe-number-validator')
+
+const schoolImpersonationService = {}
+
+/**
+ * Validate and assign school to helpdesk session
+ * @param {object} user
+ * @param {string} dfeNumber
+ * @returns {Object}
+ */
+schoolImpersonationService.validateCreateImpersonation = async (user, dfeNumber) => {
+  let validationError
+  validationError = schoolImpersonationEmptyValueValidator.validate(dfeNumber)
+  if (validationError.hasError()) {
+    return validationError
+  }
+  let school
+  try {
+    school = await schoolDataService.sqlFindOneByDfeNumber(dfeNumber)
+  } catch (error) {}
+  validationError = schoolImpersonationDfeNumberValidator.validate(school)
```

Comment:
Why do we need two validators?

Reply:
@jon-shipley the first validator checks if the value is non empty where the second checks if the dfeNumber from the client does match a school. 
In between these two validators there is a database transaction to match the school record.
The alternative would be to have one validator with two different methods.

Reply:
@jon-shipley merged to one as it seems to be cleaner.